torch/_dynamo/guards.py

```diff
@@ -100,11 +100,42 @@ def strip_getattr_getitem(name):
     return re.split(r"[.\[]", name)[0]
 
 
+@dataclasses.dataclass
```

Comment:
frozen=True?

Reply:
We mutate it

Reply:
Seems weird to do equality tests on it then 🤔

Reply:
Seems weird to mutate this too. Aren't you potentially leaking locals if you stash them on scope and then forget to free them when you're done with them?